tchannel/thrift/module.py

```diff
@@ -0,0 +1,114 @@
+from __future__ import (
+    absolute_import, division, print_function, unicode_literals
+)
+
+import inspect
+import types
+
+from .reflection import get_service_methods
+
+
+def from_thrift_module(service, thrift_module):
+
+    # start building requests instance
+    maker = ThriftRequestMaker(service, thrift_module)
+
+    # create methods that mirror thrift client
+    # and each return ThriftRequest
+    methods = _create_methods(thrift_module)
+
+    # attach to maker
+    for name, method in methods.iteritems():
+        method = types.MethodType(method, maker, ThriftRequestMaker)
+        setattr(maker, name, method)
```

Comment:
Also, I don't think this is the best approach to generating objects dynamically. You should use

    bases = (thrift_module.Iface, ThriftRequestMaker)
    type(service_name + 'RequestBuilder', bases, dct)

Where `dct` is the dictionary from method name to `create_method`-generated methods. You still get the same behavior of passing all calls through `make_request` but you're now generating a separate class for the service instead of dynamic `setattr` like that (and it now also inherits the service Iface class). You could theoretically even use `slots` or named tuple with this approach if you wanted to.